geocoding/android/src/main/java/com/baseflow/geocoding/MethodCallHandlerImpl.java

```diff
@@ -1,43 +1,63 @@
 package com.baseflow.geocoding;
 
 import android.location.Address;
+import android.os.Handler;
+import android.os.Looper;
 import android.util.Log;
+
 import androidx.annotation.Nullable;
 
 import com.baseflow.geocoding.utils.AddressMapper;
 import com.baseflow.geocoding.utils.LocaleConverter;
 
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 import io.flutter.plugin.common.BinaryMessenger;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
 
-import java.io.IOException;
-import java.util.List;
-
 /**
  * Translates incoming Geocoding MethodCalls into well formed Java function calls for {@link
  * Geocoding}.
  */
 final class MethodCallHandlerImpl implements MethodCallHandler {
     private static final String TAG = "MethodCallHandlerImpl";
     private final Geocoding geocoding;
-    @Nullable private MethodChannel channel;
+    private ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I know this works but i am also wondering if we should also consider using the TaskQueue API. https://docs.flutter.dev/development/platform-integration/platform-channels#channels-and-platform-threading

Reply:
Do you think we should use TaskQueue API? I dont have tons of experience with native android but I will try my best to implement it if you think it is worth changing to it

Reply:
I think its worth trying out as its the recommended "flutter" way